transformation/src/pipeline-config/model/pipelineConfig.ts

```diff
@@ -16,3 +16,84 @@ export interface Metadata {
   description: string;
   creationTimestamp: Date;
 }
+
+export interface PipelineConfigDTO {
+  datasourceId: number;
+  transformation: TransformationConfig;
+  metadata: MetadataDTO;
+}
+
+export interface MetadataDTO {
+  author: string;
+  displayName: string;
+  license: string;
+  description: string;
+}
+
+export class PipelineConfigDTOValidator {
+  private errors: string[] = []
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  validate (pipelineConfig: any): pipelineConfig is PipelineConfigDTO {
```

Comment:
What is the purpose of the return type `pipelineConfig is PipelineConfigDTA` when you returning a boolean?

Reply:
This construct is a [User-Defined Type Guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards).

Reply:
Thanks for the clarification. This is a neat feature.